spec/rspec/rails/matchers/have_enqueued_mail_spec.rb

```diff
@@ -0,0 +1,246 @@
+
+      expect {
+        TestMailer.email_with_optional_args('required').deliver_later
+      }.to have_enqueued_mail(TestMailer, :email_with_optional_args).with('required')
+
+      expect {
+        TestMailer.email_with_optional_args('required', 'optional').deliver_later
+      }.to have_enqueued_mail(TestMailer, :email_with_optional_args).with('required', 'optional')
+    end
+
+    it "passes with provided argument matchers" do
+      expect {
+        TestMailer.email_with_args(1, 2).deliver_later
+      }.to have_enqueued_mail(TestMailer, :email_with_args).with(1, 2)
+
+      expect {
+        TestMailer.email_with_args(1, 2).deliver_later
+      }.not_to have_enqueued_mail(TestMailer, :email_with_args).with(3, 4)
+    end
+
+    it "generates a failure message" do
+      expect {
+        expect { }.to have_enqueued_email(TestMailer, :test_email)
+      }.to raise_error(/expected to enqueue TestMailer.test_email/)
+    end
+
+    it "generates a failure message with arguments" do
+      expect {
+        expect { }.to have_enqueued_email(TestMailer, :email_with_args).with(1, 2)
+      }.to raise_error(/expected to enqueue TestMailer.email_with_args exactly 1 time with \[1, 2\], but enqueued 0/)
+    end
+
+    it "passes when deliver_later is called with a wait_until argument" do
+      send_time = Date.tomorrow.noon
+
+      expect {
+        TestMailer.test_email.deliver_later(:wait_until => send_time)
+      }.to have_enqueued_email(TestMailer, :test_email).at(send_time)
+    end
+
+    it "generates a failure message with at" do
+      send_time = Date.tomorrow.noon
+
+      expect {
+        expect {
+          TestMailer.test_email.deliver_later(:wait_until => send_time + 1)
+        }.to have_enqueued_email(TestMailer, :test_email).at(send_time)
+      }.to raise_error(/expected to enqueue TestMailer.test_email exactly 1 time at #{send_time.strftime('%F %T')}/)
+    end
+
+    it "passes when deliver_later is called with a queue argument" do
+      expect {
+        TestMailer.test_email.deliver_later(:queue => 'urgent_mail')
+      }.to have_enqueued_email(TestMailer, :test_email).on_queue('urgent_mail')
+    end
+
+    it "generates a failure message with on_queue" do
+      expect {
+        expect {
+          TestMailer.test_email.deliver_later(:queue => 'not_urgent_mail')
+        }.to have_enqueued_email(TestMailer, :test_email).on_queue('urgent_mail')
+      }.to raise_error(/expected to enqueue TestMailer.test_email exactly 1 time on queue urgent_mail/)
+    end
+
+    it "generates a failure message with unmatching enqueued mail jobs" do
+      send_time = Date.tomorrow.noon
+      queue = 'urgent_mail'
+
+      message = "expected to enqueue TestMailer.email_with_args exactly 1 time with [1, 2], but enqueued 0" + \
+                "\nQueued deliveries:" + \
+                "\n  TestMailer.test_email" + \
+                "\n  TestMailer.email_with_args with [3, 4], on queue #{queue}, at #{send_time}"
```

Comment:
Not a blocker,  but it might be nice to show other jobs are not mentioned here.

Reply:
Okay, that's a good callout; I will add that!

Reply:
👍